diesel/src/query_builder/debug_binds.rs

```diff
@@ -0,0 +1,42 @@
+use std::fmt::{self, Display};
+use std::marker::PhantomData;
+use std::mem;
+
+use backend::Backend;
+use super::{QueryFragment, AstPass};
+
+#[allow(missing_debug_implementations)]
+/// A struct that implements `fmt::Display` by walking the given AST and
+/// writing the `fmt::Debug` implementation of each bind parameter.
+pub struct DebugBinds<'a, T: 'a, DB> {
+    query: &'a T,
+    _marker: PhantomData<DB>,
+}
+
+impl<'a, T, DB> DebugBinds<'a, T, DB> {
+    pub fn new(query: &'a T) -> Self {
+        DebugBinds {
+            query,
+            _marker: PhantomData,
+        }
+    }
+}
+
+impl<'a, T, DB> Display for DebugBinds<'a, T, DB> where
+    DB: Backend,
+    T: QueryFragment<DB>,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.write_str("binds: ")?;
```

Comment:
How do you feel about treating the output as an unnamed struct? So the output would be:

```
{ query: "SELECT * FROM users WHERE id = ?", binds: [1] }
```

And then we could theoretically work properly with pretty-printing as well